### DISCUSSION TITLE
Replace function

### DISCUSSION THREAD

**Discussion by 2V3EvG4LMJFdRe** [beetbox/beets#4581] [Q&A] [answered]
_Sorry to do bother you guys here, but I asked a year ago https://github.com/beetbox/beets/discussions/3948 and https://github.com/beetbox/beets/discussions/4020 I got not answer._

### Problem

`'[<>:"\?\*\|]'` what the heck is that? I don't understand what's the syntax for `replace:` is it's not mentioned at all [on its documentation](https://beets.readthedocs.io/en/stable/reference/config.html?highlight=replace#replace).

Here's what I have so far:

```
replace:
    '…': ...
    '’': '
    '“': "
    '”': "
    '−': -
    '‐': - 
    '—': -
    'Ⅱ': II 
    'Ⅰ': I
```

My goal is to have the _tags_ written this way, don't care about how the _filenames_ are written.

**Comment by jackwilsdon** [comment, marked as answer]
`[<>:"\?\*\|]` is a [regular expression](https://en.wikipedia.org/wiki/Regular_expression) (as the documentation mentions). It matches any of the following:
* `<`
* `>`
* `:`
* `"`
* `?`
* `*`
* ` |`

The `replace` option only applies to paths and I'm not aware of any functionality in beets to allow replacing in tags - you'd need a new plugin to do that.

**Reply by 2V3EvG4LMJFdRe** [reply]
Thank you. I wonder which source has all these special characters being favored when tagging. The amount of tracks I have with the very inconvenient " or … is staggering.

**Reply by jackwilsdon** [reply]
The album-level `data_source` field should tell you where the data came from:

```
beets ls -f '$album - $title ($data_source)'
```
